docs/types/literal-suffixes.rst

```diff
@@ -0,0 +1,269 @@
+.. index:: ! literal suffix
+.. _literal_suffixes:
+
+Literal Suffixes
+================
+
+While Solidity :ref:`provides implicit conversions<types-conversion-literals>` from its literals to selected types,
+there are types for which no built-in conversions are available (most notably the
+:ref:`user-defined value types<user-defined-value-types>`).
+
+To fill this gap, it is possible to define custom conversions as *literal suffixes*.
+
+.. code-block:: solidity
+    :force:
+
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity ^0.8.20;
+
+    type Coin is uint;
+    using {add as +, eq as ==} for Coin global;
+
+    function add(Coin a, Coin b) pure returns (Coin) {
+        return Coin.wrap(Coin.unwrap(a) + Coin.unwrap(b));
+    }
+
+    function eq(Coin a, Coin b) pure returns (bool) {
+        return Coin.unwrap(a) == Coin.unwrap(b);
+    }
+
+    function c(uint mantissa, uint exponent) pure suffix returns (Coin) {
+        return Coin.wrap(mantissa * 10**6 / 10**exponent);
+    }
+
+    function uc(uint microValue) pure suffix returns (Coin) {
+        return Coin.wrap(microValue);
+    }
+
+    contract CoinSeller {
+        mapping(address => Coin) balances;
+
+        function buy() public payable {
+            require(msg.value == 1 ether);
+            assert(1.5 c == 1_500_000 uc);
+            balances[msg.sender] = balances[msg.sender] + 1.5 c;
+        }
+    }
+
+.. index:: function;call
+.. _calling_suffix_functions:
+
+Calling Suffix Functions
+------------------------
+
+There are two ways to call a suffix function: a *suffix call* and a *function call*.
+
+.. index:: ! literal suffix; suffix call syntax
+
+Suffix Call Syntax
+^^^^^^^^^^^^^^^^^^
+
+A suffix call has the same syntax as a literal with a :ref:`denomination<denominations>`:
+
+.. code-block:: solidity
+
+    42 suffix;
+    1.23 suffix;
+    0x1234 suffix;
+    'abc' suffix;
+    hex"12ff" suffix;
+    unicode"😃" suffix;
+    true suffix;
+
+The literal passed as input to the suffix function must be immediately followed by the name of the suffix.
+The two must be separated by whitespace unless it is a string, unicode or hexadecimal string literal,
+in which case the whitespace is optional (i.e. ``'abc'suffix`` is also allowed).
+
+This call syntax supports only a single literal argument.
+Variables or expressions (even as simple as wrapping the literal in parentheses) are not allowed.
+:ref:`Suffix functions defined with two parameters<suffix_function_parameters>` are also invoked with
+one literal - the :ref:`decomposition<fractional_decomposition>` of the literal into two values is
```

Comment:
```suggestion
one literal which must be of ``Integer`` type - the :ref:`decomposition<fractional_decomposition>` of the literal into two values is
```

Reply:
I guess it would be nice to make such emphases as well. But I'm approving anyway.

Reply:
I think that's irrelevant to the point I'm making here - that the suffix syntax will always take just one argument, no matter how many you may have in the definition. It's true that currently the decomposition is only done for numbers, but that's not an inherent property of this syntax. It's the consequence of what parameters suffixes can take, which will become clear when you get to the section about parameters (or follow the link).

I mean, initially the section on parameters was before the one on calls, so this particular thing was clearer, but then the issue pointed out in review was that handling of negative numbers was not clear :) I want to avoid too much repetition so in ether order there will be details you'll get only when you read both sections.